api/rs/slint/tests/simple_macro.rs

```diff
@@ -15,3 +15,36 @@ fn empty_stuff() {
     slint!(export struct Hei { abcd: bool });
     slint!(export global G { });
 }
+
+#[cfg(all(feature = "serde", feature = "serde-json"))]
+use slint::SharedVector;
+
+#[cfg(all(feature = "serde", feature = "serde-json"))]
+#[test]
+fn test_serialize_deserialize() {
+    let v = SharedVector::from([1, 2, 3]);
+    let serialized = serde_json::to_string(&v).unwrap();
+    let deserialized: SharedVector<i32> = serde_json::from_str(&serialized).unwrap();
+    assert_eq!(v, deserialized);
+}
+
+#[cfg(all(feature = "serde", feature = "serde-json"))]
+#[test]
+fn test_serialize_deserialize_person() {
```

Comment:
This is not ready, we can remove this test and instead add tests in sharedvector.rs and string.rs 
serde_json can be made a dev dependency of i-slint-core for the test.

Reply:
SharedVector has a test above .
Added tests for ShareString